core/class/jMQTTCmd.class.php

```diff
@@ -310,25 +340,72 @@ public function postSave() {
             }
 
             // Specific command : status for Broker eqpt
-            if ($this->getLogicalId() == jMQTT::CLIENT_STATUS && $this->getEqLogic()->getType() == jMQTT::TYP_BRK && $this->getEqLogic()->getIsEnable()) {
+            if ($this->getLogicalId() == jMQTT::CLIENT_STATUS && $eqLogic->getType() == jMQTT::TYP_BRK && $eqLogic->getIsEnable()) {
                 // If it's topic changed
                 if ($this->_preSaveInformations['brokerStatusTopic'] != $this->getTopic()) {
                     // Just try to remove the previous status topic
-                    $eqLogic = $this->getEqLogic();
                     $eqLogic->publish($eqLogic->getName(), $this->_preSaveInformations['brokerStatusTopic'], '', 1, 1);
                 }
             }
         }
 
-        // For info commands, check that the topic is compatible with the subscription command
-        // of the related equipment
-        if ($this->getType() == 'info' && $this->getEqLogic()->getType() == jMQTT::TYP_EQPT && !$this->getEqLogic()->getCache(jMQTT::CACHE_IGNORE_TOPIC_MISMATCH, 0)) {
-            if (! $this->topicMatchesSubscription($this->getEqLogic()->getTopic())) {
-                $this->eventTopicMismatch();
+        // For Equipments
+        if ($eqLogic->getType() == jMQTT::TYP_EQPT) {
+            // For info commands, check that the topic is compatible with the subscription command
+            if ($this->getType() == 'info' && !$eqLogic->getCache(jMQTT::CACHE_IGNORE_TOPIC_MISMATCH, 0)) {
+                if (! $this->topicMatchesSubscription($eqLogic->getTopic())) {
+                    $this->eventTopicMismatch();
+                }
             }
+			$this->listenerUpdate();
         }
     }
 
+// Listener for auto_publish action command
+	public function listenerUpdate() {
+		$cmds = array();
+		if ($this->getEqLogic()->getIsEnable() && $this->getType() == 'action' && $this->getConfiguration('auto_publish', 0)) {
+			preg_match_all("/#([0-9]*)#/", $this->getConfiguration('request', ''), $matches);
+			$cmds = array_unique($matches[1]);
+		}
+		$listener = listener::byId($this->getCache('listener', null));
```

Comment:
not sure about usage of cache : user can clean cache manually...
listener can be find by looking into 'options' (need a specific search function ...)
what's your mind about this?

Reply:
I'm not a big fan of that, but I get your point.
I have replaced getCache() by listener::searchClassFunctionOption() and removed setCache